client/components/filters/index.js

```diff
@@ -48,6 +49,7 @@ const ReportFilters = ( { advancedConfig, filters, query, path } ) => {
 		case 'advanced':
 			advancedCard = (
 				<AdvancedFilters
+					key={ queryKey }
```

Comment:
I added this so that internal state is "reset" on change of the url. The downside is the `<Search />` internal state is reset as well, losing recent search results. If we'd like to fix this, I can use `getDerivedStateFromProps` instead or keep results of the search in the data layer for quick access.